src/JuliusSweetland.OptiKey.Core/Models/XmlEyeGestures.cs

```diff
@@ -21,30 +22,28 @@ public static XmlEyeGestures ReadFromFile(string inputFilename)
 
             var gestures = new XmlEyeGestures();
             var serializer = new XmlSerializer(typeof(XmlEyeGestures));
-            try
+            
+            using (var reader = new FileStream(@inputFilename, FileMode.Open))
             {
-                using (var reader = new FileStream(@inputFilename, FileMode.Open))
-                {
-                    gestures = (XmlEyeGestures)serializer.Deserialize(reader);
-                    reader.Close();
-                }
+                gestures = (XmlEyeGestures)serializer.Deserialize(reader);                
             }
-            catch { }
-
+            
             return gestures;
         }
 
         public static XmlEyeGestures ReadFromString(string xmlString)
         {
+            if (String.IsNullOrEmpty(xmlString)) { return null; }
+
             var gestures = new XmlEyeGestures();
             var serializer = new XmlSerializer(typeof(XmlEyeGestures));
             try
             {
-                gestures = (XmlEyeGestures)serializer.Deserialize(new StringReader(xmlString));
+                return (XmlEyeGestures)serializer.Deserialize(new StringReader(xmlString));                
+            }
+            catch {
+                return null;
```

Comment:
Would it be too spammy to log an error here to help diagnose issues later?

Reply:
I think that would be fine but let me double-check